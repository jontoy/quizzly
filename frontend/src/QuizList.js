import React from "react";
import { Link } from "react-router-dom";
import QuizCard from "./QuizCard";

const QuizList = ({ quizzes }) => {
  return (
    <ul className="QuizList list-unstyled">
      {quizzes.map(({ id, name, difficulty }) => (
        <li key={id}>
          <Link to={`/quizzes/${id}`}>
            <QuizCard id={id} name={name} difficulty={difficulty} />
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default QuizList;
