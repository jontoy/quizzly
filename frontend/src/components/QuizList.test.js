import React from "react";
import { render } from "@testing-library/react";
import QuizList from "./QuizList";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <QuizList />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <QuizList />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when passed quizzes", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <QuizList
        quizzes={[{ id: "abcdefg", name: "test quiz", difficulty: 5 }]}
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
