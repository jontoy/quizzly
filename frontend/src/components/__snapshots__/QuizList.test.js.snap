// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="QuizList list-unstyled"
  />
</DocumentFragment>
`;

exports[`matches snapshot when passed quizzes 1`] = `
<DocumentFragment>
  <ul
    class="QuizList list-unstyled"
  >
    <li>
      <a
        href="/quizzes/abcdefg"
      >
        <div
          class="QuizCard card p-2 mb-1 bg-light"
        >
          <h5
            class="card-title"
          >
            test quiz
          </h5>
          <p
            class="card-text"
          >
            Difficulty: 5
          </p>
        </div>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
